annotation-utils/src/main/java/edu/isi/vista/annotationutils/PushAnnotations.kt

```diff
@@ -0,0 +1,145 @@
+    class SshConfig(val sshKeyPath: String): TransportConfigCallback {
+        override fun configure(transport: Transport?) {
+            (transport as SshTransport).sshSessionFactory = object: JschConfigSessionFactory() {
+                override fun configure(hc: OpenSshConfig.Host?, session: Session?) {}
+                override fun createDefaultJSch(fs: FS?): JSch = super.createDefaultJSch(fs).apply {
+                    addIdentity(sshKeyPath)
+
+                }
+            }
+        }
+    }
+
+    val git = Git.open(File(repoPath))
+    val config = git.getRepository().getConfig()
+    config.setString(
+            "remote",
+            "origin",
+            "url",
+            "git@github.com:isi-vista/curated-training-annotation.git")
+    config.save()
+
+    val currentDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+    val formatter = SimpleDateFormat("yyyy-MM-dd-HH-mm-ss")
+
+    // Currently working on test_branch
+    git.checkout().setName("test_branch").call()
+
+    git.add().addFilepattern("data/").call()
+    val newCommit = git.commit()
+            .setMessage("includes annotations from ${currentDate.getTime()}").call()
+    git.tag().setName(formatter.format(currentDate.getTime())).call()
+
+
+    if (hasDiff(git, newCommit)) {
+        val pushIter = git.push()
+                .setTransportConfigCallback(SshConfig("~/.ssh/id_rsa"))
+                .call()
+        val pushResult = pushIter.iterator().next()
+    } else {
+        println("No new annotations - nothing will be pushed.")
+        git.reset().setRef("HEAD~1").setMode(ResetCommand.ResetType.SOFT).call()
+    }
+}
+
+// Compare old data to newly exported data
+fun hasDiff(git: Git, newCommit: RevCommit): Boolean {
+    val oldCommit = getPrevCommit(git, newCommit)
+    if (oldCommit == null) {
+        println("This is the first commit.")
+        return true
+    }
+    val oldTree = getCanonicalTreeParser(git, oldCommit)
+    val newTree = getCanonicalTreeParser(git, newCommit)
+    val diffs: List<DiffEntry> =
+            git.diff().setNewTree(newTree).setOldTree(oldTree).call()
+    if (diffs.isEmpty()) {
+        return false
+    }
+    return true
+}
+
+fun getPrevCommit(git: Git, commit: RevCommit): RevCommit? {
+    val walk = RevWalk(git.getRepository())
+    walk.markStart(commit)
+    var count = 0
+    for (rev in walk) {
+        if (count ==1) {
+            return rev
+        }
+        count = count.inc()
+    }
+    return null
+}
+
+fun getCanonicalTreeParser(git: Git, commit: ObjectId): CanonicalTreeParser {
+    val walk = RevWalk(git.getRepository())
+    val revCommit = walk.parseCommit(commit)
+    val treeId = revCommit.getTree().getId()
+    val reader = git.getRepository().newObjectReader()
+    return CanonicalTreeParser(null, reader, treeId)
+}
+
+
+fun main(argv: Array<String>) {
```

Comment:
As noted above, put this function first so the reader can get a high-level overview of what the program is doing before diving into details

Reply:
Done